flow-decomposition/src/main/java/com/powsybl/flow_decomposition/FlowDecompositionComputer.java

```diff
@@ -47,6 +50,11 @@ public FlowDecompositionComputer(FlowDecompositionParameters flowDecompositionPa
                                      String loadFlowProvider, String sensitivityAnalysisProvider) {
         this.parameters = flowDecompositionParameters;
         this.loadFlowParameters = loadFlowParameters;
+        if (!LoadFlowParameters.ConnectedComponentMode.MAIN.equals(loadFlowParameters.getConnectedComponentMode())) {
+            LOGGER.warn("Flow decomposition is currently available only on the main synchronous component. Changing connected component mode from {} to MAIN.",
+                    loadFlowParameters.getConnectedComponentMode());
+            loadFlowParameters.setConnectedComponentMode(LoadFlowParameters.ConnectedComponentMode.MAIN);
+        }
```

Comment:
Could you copy the `loadFlowParameters` to avoid side effects ?

Reply:
Good suggestion